modules/aggregation-layers/src/heatmap-layer/heatmap-layer.js

```diff
@@ -118,8 +122,15 @@ export default class HeatmapLayer extends CompositeLayer {
       this._updateTextureRenderingBounds();
     }
 
-    this.setState({zoom: opts.context.viewport.zoom});
+    if (oldProps.colorDomain !== props.colorDomain || changeFlags.viewportChanged) {
+      const {viewport} = this.context;
+      const domainScale = viewport ? 1024 / viewport.scale : 1;
+      newState.colorDomain = props.colorDomain.map(x => x * domainScale);
```

Comment:
- Check `props.colorDomain` and fall back to `DEFAULT_COLOR_DOMAIN` here
- Just directly call `setState` here. Creating and merging the `newState` object is more expensive.